### PR TITLE
Rewardpointsbug

### PR DESCRIPTION
**Pull request by tienquocnguyen**
users should now have reward points being added back to their account upon cancel outside of the 48 hours range.

### CODE REVIEW

File: routes/api/bookin.js [hunk 1]
```diff
@@ -467,14 +467,20 @@ router.post('/changeReservation',(req,res)=>{
...
-            if (booking.status == 0){
-                booking.status = 3;
-                if (booking.rewardPointsUsed !== 0){
-                    User.findOneAndUpdate(
-                        {'customerID': booking.customerID},
-                        {$inc: { "rewardPoints" : booking.rewardPointsUsed }
-                    })
+            if (booking.status === 0){
+                booking.status = 0;
+                if (booking.check_in_date - new Date > 172800000){
```
Comment (HongbinZheng) [inline routes/api/bookin.js]: Should also check if user had make a change to their reservation, if they have, there will be anew check in date

**Comment by ynoe** [comment]
Seems logical, however, the rewards point refunded is not reflected on /history page..

**Comment by SandroSal** [comment]
just so we are all on the same page...

when a user makes a NEW booking, rewards points are not directly added to the account. He/She has to wait till checkin. after successful checkin, the rewards points are added to the total balance.
if a user uses rewardspoints to pay for the trip, they will be immediately deducted from the balance.


If a user changes/cancels a booking:
- rewardspoints EARNED will be updated, but there won't be any change to the total balance
- rewardspoints USED will also be updated, and there will be an immediate change to the total balance (either addition or deduction)

@HongbinZheng  please confirm this
@tienquocnguyen please confirm this
@ynoe please confirm this

**Comment by tienquocnguyen** [comment]
Confirm

**Comment by ynoe** [comment]
yes

**Comment by HongbinZheng** [comment]
> Seems logical, however, the rewards point refunded is not reflected on /history page..

It do reflect the rewards points refunded, can you run one more test?

**Comment by HongbinZheng** [comment]
Confirm

**Comment by SandroSal** [comment]
I just tested it... rewards points were not added back

**Comment by SandroSal** [comment]
rewardPointsUsed is also not updated

**Comment by SandroSal** [comment]
I added some code that will show the rewardsPointsUsed in the expansionpanel... that might help with debugging ... please leave to code in anyways, as it is a nice feature for the user too

**Comment by SandroSal** [comment]
okay... so I thought this PR was for cancel and change...

when cancelling, all points are refunded. that works

changing however does not refund to total, nor update rewardsPointsUsed

**Comment by HongbinZheng** [comment]
fixed the rewardsPointsUsed problem, please test it

**Pull request merged**